e2e/testcases/cli_test.go

```diff
@@ -833,3 +845,192 @@ func TestNomosVetNamespaceRepo(t *testing.T) {
 		})
 	}
 }
+
+func TestCLIBugreportNomosRunningCorrectly(t *testing.T) {
+	var bugReportZipName, bugReportDirName string
+	tmpDir := nomostest.TestDir(t)
+	tw := nomostesting.New(t, nomostesting.NomosCLI)
+	opts := nomostest.NewOptStruct(nomostest.TestClusterName(tw), tmpDir, tw)
+	nomostest.FreshTestEnv(t, opts)
+
+	// get bugreport
+	out, err := exec.Command("nomos", "bugreport").CombinedOutput()
```

Comment:
You need to switch to the test context, instead of the current context.

You can do so by two steps:
1. use the kubectl command to get the current test context: `nt.Kubectl*config", "current-context"), which takes `nt.kubeconfigPath` as `--kubeconfig`.
2. use the kubectl command to switch to the context: `nt.MustKubectl("config", "use-context", testContext)`

Then `nomos bugreport` is guaranteed to run with the test context, instead of the default one.

Reply:
The test seems to have no issue with recognizing context, if it is for the error mentioned in [this comment](https://github.com/GoogleContainerTools/kpt-config-sync/pull/134#discussion_r998807686). I've tested it in both GKE and Kind.

Reply:
As discussed in the mentioned thread, setting kubectl context turns out to be not relevant in this scenario.